File: 05.1 LayoutTest/src/net/terzeron/layouttest/MainActivity.java
package net.terzeron.layouttest;

import android.os.Bundle;
import android.view.Menu;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.View.OnLongClickListener;
import android.widget.Button;
import android.widget.Toast;
import android.app.Activity;
import android.graphics.Color;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
      
        Button button1 = (Button) findViewById(R.id.button1);
        button1.setOnClickListener(new OnClickListener() {
        	public void onClick(View v) {
        		Toast.makeText(MainActivity.this, ((Button) v).getText(), Toast.LENGTH_SHORT).show();
        	}
        });
        
        Button button2 = (Button) findViewById(R.id.button2);
        button2.setOnLongClickListener(new OnLongClickListener() {
        	public boolean onLongClick(View v) {
        		Button b = (Button) findViewById(R.id.button2);
        		b.setBackgroundColor(Color.RED);
        		return false;
        	}
        });
        
    }
    
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }
    
}

File: 06.EventTest/app/src/main/java/net/terzeron/a06eventtest/MainActivity.kt
package net.terzeron.a06eventtest

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        myButton.setOnClickListener {
            statusText.text = "Button clicked"
        }

        myButton.setOnLongClickListener {
            statusText.text = "Long button clicked"
            true
        }

    }
}
